sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/autocommit/AutocommitMLS.java

```diff
@@ -35,42 +35,45 @@ public void processRecords(ConsumerRecords<K, V> records, KafkaConsumer<K, V> co
       consumerName = KafkaHelper.getClientId(consumer);
     }
 
-    for (ConsumerRecord<K, V> record : records) {
-      LOGGER.debug("Handling message for {}", record.key());
-      try {
-        SimpleTimer timer = new SimpleTimer();
-        handler.handle(record);
-        addOffsetToCommitOnClose(record);
+    for (ConsumerRecord<K, V> consumerRecord : records) {
+      LOGGER.debug("Handling message for {}", consumerRecord.key());
+      try (var ignored = messageHandlerContextFor(consumerRecord)) {
+        try {
```

Comment:
Can we merge these two try statements?

Reply:
No. Tests are failing then. We [would loose the context for error handling](https://stackoverflow.com/a/25057349). I'm coding Java for decades now but I'm still learning.

Reply:
Wow. Also did not know that. 👍